module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,53 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& arr) {
+    std::vector<std::pair<uint8_t, uint8_t>> vec;
+    vec.reserve(width * height);
+    uint8_t buffer;
+    std::for_each(arr.begin(), arr.end(), [&vec, &buffer](auto ln) {
+        buffer = ln.front();
+        auto it1 = ln.begin();
+        while (it1 != ln.end()) {
+            auto it2 = std::find_if(it1, ln.end(), [&buffer](const auto a) { return buffer != a; });
+            vec.push_back(std::make_pair(buffer, std::count(it1, it2, buffer)));
```

Comment:
instead std::make_pair you can use:
```suggestion
            vec.push_back( {(buffer, std::count(it1, it2, buffer)) } );
```

Reply:
Isn't make_pair more clear and easier to read? Or is there some drawback about make_pair I don't know about? :D

Reply:
I don't know anything about flaws but it's only my suggestion :)

Reply:
There is thread about this in stackoverflow:
https://stackoverflow.com/questions/53900865/emplace-back-vs-push-back-when-inserting-a-pair-into-stdvector